### DISCUSSION TITLE
How Trino can show Chinese in the query results support?

### DISCUSSION THREAD

**Discussion by mana6666** [trinodb/trino#23259] [General]
presto verson : 302

cannot get the chinese result in correct form according to issue#3517.
when i input the sql below :select to_utf8(return_chinese_col) from table limit 1.
the return is :
 e7 91 9e e6 98 8c e6 a8 aa e6 b8 af e7 94 b5 e4 
 bf a1 e6 89 80 e7 bb bc e5 90 88 e6 9c ba e6 88 
 bf 2f e5 8d 8e e4 b8 ba e6 8a 80 e6 9c af e6 9c 
 89 e9 99 90 e5 85 ac e5 8f b8 4d 41 35 36 38 30 
 54 2d 4f 4c 54 2f 4a 4a 2d 4f 4c 54 30 30 30 31 
 5f 5f 5f 30 2f 34 2f 30  
not chinese wordk.

**Comment by mana6666** [comment]
when i input the sql below :select from_utf8(to_utf8(en)) from pon_min_partitioned limit 1;
presto:mana> select from_utf8(to_utf8(en)) from pon_min_partitioned limit 1;
                      _col0                       
--------------------------------------------------
 <E8><B4><B5><E9><98><B3><E5><B8><82>-<E4><BA><91><E5><B2><A9>682<E5><B1><80>-<E7><BB><BC><E5><90><88><E6><9C><BA><E6><88><BF>-OLT-H9-MA5680T-G1030301___0-12-4 
(1 row)

**Comment by dain** [comment]
The [`to_utf8`](https://prestosql.io/docs/current/functions/string.html#to_utf8) function converts a VARCHAR (text) value to a VARBINARY (binary) value.  Instead you should be able to just use `SELECT return_chinese_col FROM table LIMIT 1`.

Also, 302 is very, very, old.  I suggest you try with the latest release (currently 340) https://prestosql.io/download.html

**Comment by mana6666** [comment]
i get the result below: 
????Z01
if i input the sql :SELECT return_chinese_col FROM table LIMIT 1.
on the verson 302 in the docker download from https://prestosql.io/download.html

**Comment by ssquan** [comment]
Hi, @mana6666  Just uses `select` works fine even in the old version for me. Presto can't show Chinese correctly if the real charset of data is not the claimed one, this may happen if you sync data from RMDB to hive incorrectly and then read it by Presto. But it's not a Presto issue.

**Comment by shanying2017** [comment]
@ssquan thanks a lot
we use the alluxio+ hive +presto docker image with the default configuration, but I can get the result from hive cli well
https://hub.docker.com/r/alluxio/alluxio-presto-sandbox/tags

this is the hive.properties
connector.name=hive-hadoop2
hive.metastore.uri=thrift://localhost:9083
hive.hdfs.impersonation.enabled=false
hive.recursive-directories=true

so ,  where is wrong? about thrift ?
